app/assets/javascripts/components/NotesList.tsx

```diff
@@ -0,0 +1,118 @@
+import { KeyboardKey } from '@/services/ioService';
+import { AppState } from '@/ui_models/app_state';
+import { DisplayOptions } from '@/ui_models/app_state/notes_view_state';
+import { SNNote } from '@standardnotes/snjs';
+import { observer } from 'mobx-react-lite';
+import { FunctionComponent } from 'preact';
+import { NotesListItem } from './NotesListItem';
+import { toDirective } from './utils';
+
+type Props = {
+  appState: AppState;
+  notes: SNNote[];
+  selectedNotes: Record<string, SNNote>;
+  displayOptions: DisplayOptions;
+  paginate: () => void;
+};
+
+const FOCUSABLE_BUT_NOT_TABBABLE = -1;
+const NOTES_LIST_SCROLL_THRESHOLD = 200;
+
+export const NotesList: FunctionComponent<Props> = observer(
+  ({ appState, notes, selectedNotes, displayOptions, paginate }) => {
+    const { selectPreviousNote, selectNextNote } = appState.notesView;
+    const { hideTags, hideDate, hideNotePreview, sortBy } = displayOptions;
+
+    const tagsStringForNote = (note: SNNote): string => {
+      if (hideTags) {
+        return '';
+      }
+      const selectedTag = appState.selectedTag;
+      if (!selectedTag) {
+        return '';
+      }
+      if (selectedTag?.isSmartTag) {
```

Comment:
Can you remove the `?` check here?

Typescript should be smart enough to know there is a value (since we would have exited the code with the return before),
if it doesn't then we might have missed a case.